### PR TITLE
Merge Everest into ert

### PR DESCRIPTION
**Pull request by oyvindeide**
**Issue**
Closes https://github.com/equinor/everest/issues/67

One question is how to deal with the version, currently I checked in the version file, and we would track it manually.

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Make sure unit tests pass locally after every commit (`git rebase -i 10
      --exec 'pytest tests/unit_tests -n logical -m "not integration_test"'`)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: .github/workflows/run-tests-build-docs.yml
Comment (yngve-sk) [inline .github/workflows/run-tests-build-docs.yml:1]: Should we maybe rename the ERT/Everest workflows a bit to clarify which is Everest and which is ERT?

File: src/everest/version.py
Comment (yngve-sk) [inline src/everest/version.py:2]: Is this supposed to be here?

**Comment by yngve-sk** [comment]
Maybe future work / out of scope right now, so for now just an open question: How should we run tests on PRs? I guess now we just run all of it for all PRs.

Some different scenarios
1) PR with a change only in `src/ert` and/or `tests/ert`
2) PR with a change only in `src/everest` and/or `tests/everests`, test
3) PR with changes to both ERT and Everest (I guess here we just run everything)

File: tests/everest/conftest.py
Comment (yngve-sk) [inline tests/everest/conftest.py]: From the diff, a lot of stuff was removed from this file. Where did it go? Is it now mostly imported from ERT? All the qt related fixtures for example. Not  obvious from the diff, guessing it is now also getting that from ERT?

File: examples/egg/eclipse/include/EGG.DATA
Comment (yngve-sk) [inline examples/egg/eclipse/include/EGG.DATA:1]: Do we want the `everest` `examples` folder to be called `examples` still? Same for test-data / test_data, could perhaps do some more explicit folder naming for these? (Maybe out of scope of this PR, but there are also some examples embedded within the everest test folder, `everest/tests/test_data).

Comment (frode-aarstad) [reply examples/egg/eclipse/include/EGG.DATA:1]: I second the above.

File: .github/workflows/run-tests-build-docs.yml
Comment (frode-aarstad) [reply .github/workflows/run-tests-build-docs.yml:1]: I second the above. Makes it easier in the "checks" part the PRs to see which part failed.

**Comment by frode-aarstad** [comment]
Very good. IMO its mergable as is. 

We will just need to monitor closely the next few weeks how everything works and fix issues as they show up

File: .github/workflows/run-tests-build-docs.yml
Comment (DanSava) [reply .github/workflows/run-tests-build-docs.yml:1]: The name of the workflow files should also be changed to make it clear if they are from ERT or Everest

File: .github/workflows/build_and_test.yml
Comment (frode-aarstad) [inline .github/workflows/build_and_test.yml:139]: Are there a similar test for everest docs?

**Comment by yngve-sk** [comment]
Overall LGTM, imo mergeable after fixing up some smaller things:
* naming/structure stuff from comments above, 
* working build of the keyword reference docs
* [`test_init_no_project_res` is skipped when running locally without everest models installed, or everest models is added as a test dependency].

Whatever remains can be fixed post-merge, not expecting/seeing too many obvious potential problems that can't be discovered/fixed after the merge

**Comment by frode-aarstad** [comment]
Closes https://github.com/equinor/everest/issues/121

File: .github/workflows/build_and_test.yml
Comment (oyvindeide) [reply .github/workflows/build_and_test.yml:139]: Yes, it is in the everest workflow, renamed it to `test_everest.yml`

File: .github/workflows/run-tests-build-docs.yml
Comment (oyvindeide) [reply .github/workflows/run-tests-build-docs.yml]: Renamed them now

File: src/everest/version.py
Comment (oyvindeide) [reply src/everest/version.py]: Good point, I removed everything related to this, and just use the ert version

**Comment by oyvindeide** [comment]
> Maybe future work / out of scope right now, so for now just an open question: How should we run tests on PRs? I guess now we just run all of it for all PRs.
>
> Some different scenarios
>
>    PR with a change only in src/ert and/or tests/ert
>    PR with a change only in src/everest and/or tests/everests, test
>    PR with changes to both ERT and Everest (I guess here we just run everything)

This is a good point, this sets it up to do everything, while it is possible to filter so that some workflows only trigger on changes in a sub-directory, if that workflow is required, it will be marked as pending if it does not start, and will block merging. There are options to bypass this, but they come with their own complications.

File: examples/egg/eclipse/include/EGG.DATA
Comment (oyvindeide) [reply examples/egg/eclipse/include/EGG.DATA:1]: I agree with these changes, though could also see us creating issues for the post-merge, as I suspect it will be some work moving these files around.

File: tests/everest/conftest.py
Comment (oyvindeide) [reply tests/everest/conftest.py]: It is mostly a git thing, as the `conftest.py` from ert and everest are very similar, so it thinks this was moved and then deleted (confusing it with the `ert` conftest.py. Will see if I can improve the diff.

**Review by frode-aarstad — APPROVED** [review]
Good job! 🚀

File: .github/workflows/test_everest_macos.yml
Comment (yngve-sk) [inline .github/workflows/test_everest_macos.yml]: This seems to undo the prev commit of only running Mac tests on push to everest paths, intentional?

Comment (oyvindeide) [reply .github/workflows/test_everest_macos.yml]: It was intenetional, because it already just runs on tags, from below:

```yaml
if: github.ref_type == 'tag' # only build for mac when tags
```

**Comment by roliveira** [comment]
🚀

**Pull request merged**
